src/pudl/transform/ferc1.py

```diff
@@ -6322,6 +6266,100 @@ def table_to_column_to_check() -> dict[str, list[str]]:
+
+    This data isn't incorrect, it just interferes with how we process the calculations
+    embedded within these tables. This is why we are applying this within
+    :func:`_core_ferc1__table_dimensions` because that is what we use to build the
+    calculation components table.
+    """
+    # made these variables so we could generalize to other tables if we desired that.
+    xbrl_factoid = "xbrl_factoid"
+    dimension_col = "utility_type"
+    idx = ["utility_id_ferc1", "report_year", xbrl_factoid]
+
+    # first we need to find the xbrl_factoid's where there are mostly only totals
+    # within the dimension_col. Which is to say there are a few rare non-total
+    # utility_type's.
+    tot_mask = df[dimension_col] == "total"
+    # build a dataframe of xbrl_factoid's with a column for the count of the records
+    # with utility_type total and another column with the count for all the other
+    # utility_type's
+    util_type_count = pd.merge(
+        pd.DataFrame(df.loc[tot_mask, xbrl_factoid].value_counts()),
+        pd.DataFrame(df.loc[~tot_mask, xbrl_factoid].value_counts()),
+        on=xbrl_factoid,
+        how="outer",
+        suffixes=("_total", "_other"),
+    )
+    # With that little xbrl_factoid count dataframe we can check to see which
+    # xbrl_factoid's are almost entriely total. In order to id these "rare"
+    # non-totals we use a threshold of 20% here. meaning only 20% of the instances
+    # of the utility_type for a particular xbrl_factoid were non-totals.
+    # that's a little bit arbitrary... This was tested up to 40% but that seems
+    # too aggressive.
+    mostly_total_xbrl_factoids = util_type_count[
+        util_type_count.count_other.notnull()
+        & (
+            (
+                util_type_count.count_other
+                / (util_type_count.count_other + util_type_count.count_total)
+            )
+            < 0.40
+        )
+    ].index
+    logger.info(
+        f"{len(mostly_total_xbrl_factoids) / len(df[xbrl_factoid].unique()):.1%} of the "
+        f"xbrl_factoid's in this table have {dimension_col} values that are mostly total, "
+        "but have some rare instances of non-total values."
+    )
+    # add a utility type count column. Because we only care about this when there
+    # are more than one util type
+    df.loc[:, "util_type_count"] = df.groupby(idx)[[dimension_col]].transform("count")
+    mostly_totals_mask = df[xbrl_factoid].isin(
+        [fact for fact in mostly_total_xbrl_factoids if "correction" not in fact]
+    ) & (df.util_type_count > 1)
+
+    mostly_totals_idx = (
+        df.loc[mostly_totals_mask, idx].drop_duplicates().set_index(idx).index
+    )
+    mixed_typed_income = df.set_index(idx).loc[mostly_totals_idx].reset_index()
+    # first remove the records with null non-total records
+    maybe_unique = mixed_typed_income[
+        ~(mixed_typed_income[dimension_col] != "total")
```

Comment:
Is there a reason for the double negative here?  Why `~(x != "total")` rather than `(x == "total")`?  Or were you trying to negate the combination of the two conditions?  If we want to drop all the records where dollar value is null, and we have a non-total dimension, would it be:

```py
maybe_unique = mixed_type_income[
    ~(
        (mixed_typed_income[dimension_col] != "total")
        & (mixed_typed_income["dollar_value"].isna())
    )
]
```